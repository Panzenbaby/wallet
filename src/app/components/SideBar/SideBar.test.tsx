import React from "react";
import { renderWithRouter } from "testing-library";

import { SideBar } from "./SideBar";

describe("SideBar", () => {
	it("should render empty", () => {
		const { container, asFragment } = renderWithRouter(<SideBar />);

		expect(container).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with items", () => {
		const items = [
			{
				icon: "Sliders",
				key: "general",
				label: "General",
			},
			{
				icon: "Lock",
				key: "password",
				label: "Password",
			},
			{
				icon: "ArrowUpTurnBracket",
				key: "export",
				label: "Export",
			},
		];

		const { container, asFragment, getAllByRole } = renderWithRouter(<SideBar items={items} />);

		expect(container).toBeTruthy();
		expect(getAllByRole("listitem").length).toEqual(3);
		expect(asFragment()).toMatchSnapshot();
	});
});
