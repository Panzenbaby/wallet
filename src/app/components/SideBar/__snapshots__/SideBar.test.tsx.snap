// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render empty 1`] = `
<DocumentFragment>
  <div
    class="SideBar__SideBarContainer-sc-1wg5vvh-0 dinsck"
  >
    <ul />
  </div>
</DocumentFragment>
`;

exports[`SideBar should render with items 1`] = `
<DocumentFragment>
  <div
    class="SideBar__SideBarContainer-sc-1wg5vvh-0 dinsck"
  >
    <ul>
      <div
        class="relative cursor-pointer"
        data-testid="side-menu__item--undefined"
      >
        <li
          class="rounded py-5 pl-10 pr-8 h-15 flex items-center text-theme-primary-600 dark:text-theme-secondary-200 bg-theme-primary-100 dark:bg-theme-secondary-800"
        >
          <div
            class="absolute top-0 bottom-0 left-0 w-2 rounded bg-theme-primary-600"
          />
          <div
            class="mr-3"
          >
            <div
              class="sc-bdnxRM NjstW"
              height="20"
              width="20"
            >
              <svg>
                sliders.svg
              </svg>
            </div>
          </div>
          <span
            class="text-lg font-semibold"
          >
            General
          </span>
        </li>
      </div>
      <div
        class="relative cursor-pointer"
        data-testid="side-menu__item--undefined"
      >
        <li
          class="rounded py-5 pl-10 pr-8 h-15 flex items-center text-theme-primary-600 dark:text-theme-secondary-200 bg-theme-primary-100 dark:bg-theme-secondary-800"
        >
          <div
            class="absolute top-0 bottom-0 left-0 w-2 rounded bg-theme-primary-600"
          />
          <div
            class="mr-3"
          >
            <div
              class="sc-bdnxRM NjstW"
              height="20"
              width="20"
            >
              <svg>
                lock.svg
              </svg>
            </div>
          </div>
          <span
            class="text-lg font-semibold"
          >
            Password
          </span>
        </li>
      </div>
      <div
        class="relative cursor-pointer"
        data-testid="side-menu__item--undefined"
      >
        <li
          class="rounded py-5 pl-10 pr-8 h-15 flex items-center text-theme-primary-600 dark:text-theme-secondary-200 bg-theme-primary-100 dark:bg-theme-secondary-800"
        >
          <div
            class="absolute top-0 bottom-0 left-0 w-2 rounded bg-theme-primary-600"
          />
          <div
            class="mr-3"
          >
            <div
              class="sc-bdnxRM NjstW"
              height="20"
              width="20"
            >
              <svg>
                arrow-up-turn-bracket.svg
              </svg>
            </div>
          </div>
          <span
            class="text-lg font-semibold"
          >
            Export
          </span>
        </li>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
